spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -814,7 +821,17 @@ public Object receiveAndConvert() throws AmqpException {
 
 	@Override
 	public Object receiveAndConvert(String queueName) throws AmqpException {
-		Message response = receive(queueName);
+		return receiveAndConvert(queueName, this.receiveTimeout);
+	}
+
+	@Override
+	public Object receiveAndConvert(long timeoutMillis) throws AmqpException {
+		return receiveAndConvert(this.getRequiredQueue(), timeoutMillis);
+	}
+
+	@Override
+	public Object receiveAndConvert(String queueName, long timeoutMillis) throws AmqpException {
+		Message response = receive(queueName, timeoutMillis);
```

Comment:
Doh. I didn't test the entire project with hope to take a look to the Travis afterwards.
But as you see I forgot that :cry: 

Have you fixed everything already?
Or can I push it?

Thank you anyway!